logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputStrategyExt.java

```diff
@@ -225,12 +224,23 @@ protected SimpleAbstractOutputStrategyExt(final Ruby runtime, final RubyClass me
             super(runtime, metaClass);
         }
 
-        @JRubyMethod(required = 4)
+        @JRubyMethod(required = 5)
         public IRubyObject initialize(final ThreadContext context, final IRubyObject[] args) {
             final RubyClass outputClass = (RubyClass) args[0];
             // Calling "new" here manually to allow mocking the ctor in RSpec Tests
-            output = args[0].callMethod(context, "new", args[3]);
+            output = args[0].callMethod(context, "new", args[4]);
             initOutputCallsite(outputClass);
+
+            final RubyClass codecDelegatorClass = (RubyClass) RubyUtil.RUBY.executeScript(
+                    "require 'logstash/codecs/delegator'\nLogStash::Codecs::Delegator",
```

Comment:
regarding this pattern - its fine in tests but ideally should not be used in production code.
... there's seems to be a pattern of keeping `RubyClass` instances under `RubyUtil` // *this is preferred, right?*
2nd option would be to have the require happen once (elsewhere) and than just retrieve the constant
e.g. `RUBY.getClassFromPath("LogStash::Codecs::Delegator")`